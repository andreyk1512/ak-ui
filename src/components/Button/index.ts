export * from './Button';
export * from './Button.types';

import { Link } from './components';
import { Button as OriginButton } from './Button';

export type ButtonProps = typeof OriginButton & {
  Link: typeof Link;
};

export const Button = OriginButton as ButtonProps;
Button.Link = Link;
