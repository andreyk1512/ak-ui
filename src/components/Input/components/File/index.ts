export * from './File';
export * from './File.types';
