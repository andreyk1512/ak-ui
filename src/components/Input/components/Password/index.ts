export * from './Password';
export * from './Password.types';
