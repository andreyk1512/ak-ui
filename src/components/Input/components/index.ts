export * from './Password';
export * from './TextArea';
export * from './File';
