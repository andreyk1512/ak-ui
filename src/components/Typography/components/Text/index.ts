export * from './Text.tsx';
export * from './Text.types';
