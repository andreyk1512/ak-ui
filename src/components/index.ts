export * from './Tag';
export * from './Button';
export * from './Checkbox';
