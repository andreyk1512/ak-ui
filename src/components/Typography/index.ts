export * from './Typography';
export * from './Typography.types';

import { Text, Title, Paragraph } from './components';
import { Typography as OriginTypography } from './Typography';

export type TypographyProps = typeof OriginTypography & {
  Text: typeof Text;
  Title: typeof Title;
  Paragraph: typeof Paragraph;
};

export const Typography = OriginTypography as TypographyProps;
Typography.Text = Text;
Typography.Title = Title;
Typography.Paragraph = Paragraph;
