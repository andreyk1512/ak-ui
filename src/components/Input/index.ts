export * from './Input';
export * from './Input.types';

import { Password, TextArea, File } from './components';
import { Input as OriginInput } from './Input';

export type InputProps = typeof OriginInput & {
  Password: typeof Password;
  TextArea: typeof TextArea;
  File: typeof File;
};

export const Input = OriginInput as InputProps;
Input.Password = Password;
Input.TextArea = TextArea;
Input.File = File;
