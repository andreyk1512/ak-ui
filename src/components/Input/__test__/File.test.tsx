import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { File } from '../components';

describe('Input.File', () => {
  it('should render the file input element', () => {
    const placeholder = 'select';
    const { container } = render(<File placeholder={placeholder} />);

    const input = screen.getByLabelText(placeholder);
    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute('type', 'file');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should display name for selected file', () => {
    const file = { name: "TestFileName" } as File;
    const { container } = render(<File value={file} />);

    const input = screen.getByLabelText(file.name);
    expect(input).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should reset value on cross icon click', () => {
    const file = { name: "TestFileName" } as File;
    const onChange = jest.fn();
    render(<File value={file} onChange={onChange} />);

    const crossIcon = screen.getByLabelText('remove-icon');
    fireEvent.click(crossIcon);
    expect(onChange).toHaveBeenCalledWith(null);
  });
});
