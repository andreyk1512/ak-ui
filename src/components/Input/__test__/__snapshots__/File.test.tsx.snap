// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.File should display name for selected file 1`] = `
<label
  class="css-nc1f5d"
>
  <input
    class="css-38lglc"
    type="file"
  />
  <span />
  <span
    class="css-av1h6s"
  >
    TestFileName
  </span>
  <span
    aria-label="remove-icon"
    class="css-5bno9x"
    height="12"
    width="12"
  />
</label>
`;

exports[`Input.File should render the file input element 1`] = `
<label
  class="css-nc1f5d"
>
  <input
    class="css-38lglc"
    type="file"
  />
  <span />
  <span
    class="css-av1h6s"
  >
    select
  </span>
</label>
`;
